openshift-kube-apiserver/admission/customresourcevalidation/ingress/validate_ingress.go

```diff
@@ -0,0 +1,156 @@
+package ingress
+
+import (
+	"fmt"
+	"io"
+
+	configv1 "github.com/openshift/api/config/v1"
+	"github.com/openshift/library-go/pkg/route/routeapihelpers"
+	"k8s.io/apimachinery/pkg/api/validation"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/apimachinery/pkg/util/sets"
+	"k8s.io/apimachinery/pkg/util/validation/field"
+	"k8s.io/apiserver/pkg/admission"
+	"k8s.io/kubernetes/openshift-kube-apiserver/admission/customresourcevalidation"
+)
+
+const PluginName = "config.openshift.io/ValidateIngress"
+
+// Register registers a plugin
+func Register(plugins *admission.Plugins) {
+	plugins.Register(PluginName, func(config io.Reader) (admission.Interface, error) {
+		return customresourcevalidation.NewValidator(
+			map[schema.GroupResource]bool{
+				configv1.Resource("Ingresses"): true,
```

Comment:
Maybe it doesn't matter, but shouldn't the resource name be lower-case?
```suggestion
				configv1.Resource("ingresses"): true,
```